extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcBasePersistenceImpl.java

```diff
@@ -417,9 +415,17 @@ public <T> List<T> listEntities(
     // absence of transaction.
     String query = QueryGenerator.generateSelectQuery(ModelEntity.class, params);
     try {
-      List<PolarisBaseEntity> results =
-          datasourceOperations.executeSelect(
-              query, ModelEntity.class, ModelEntity::toEntity, entityFilter, limit);
+      List<PolarisBaseEntity> results = new ArrayList<>();
+      datasourceOperations.executeSelectOverStream(
+          query,
+          new ModelEntity(),
+          stream -> {
+            stream
+                .map(ModelEntity::toEntity)
+                .filter(entityFilter)
+                .limit(limit)
+                .forEach(results::add);
```

Comment:
is there a way to do `toList()` ?

Reply:
With the loan pattern, we aren't returning anything so we need to get the list as a side effect. Unfortunately, there is a restriction in Java that variables pulled in to the scope of a lambda must be "effectively final", so we can't do like `results = ...toList()` in the lambda